### ISSUE TITLE
training_epoch_loop.py", line 486, in _update_learning_rates     raise MisconfigurationException( pytorch_lightning.utilities.exceptions.MisconfigurationException: ReduceLROnPlateau conditioned on metric val/forecast_loss which is not available. Available metrics are: ['train/mape', 'train/mae', 'train/mse', 'train/rse', 'train/forecast_loss', 'train/class_loss', 'train/loss', 'train/acc']. Condition can be set using `monitor` key in lr scheduler dict

### ISSUE THREAD

**Issue by jeevanu** [QData/spacetimeformer#17]
training_epoch_loop.py", line 486, in _update_learning_rates
    raise MisconfigurationException(
pytorch_lightning.utilities.exceptions.MisconfigurationException: ReduceLROnPlateau conditioned on metric val/forecast_loss which is not available. Available metrics are: ['train/mape', 'train/mae', 'train/mse', 'train/rse', 'train/forecast_loss', 'train/class_loss', 'train/loss', 'train/acc']. Condition can be set using `monitor` key in lr scheduler dict

**Comment by dialuser** [comment]
I got the same error. Have you resolved it? 
pytorch_lightning.utilities.exceptions.MisconfigurationException: ReduceLROnPlateau conditioned on metric val_checkpoint_on which is not available. Available metrics are: ['train/mape', 'train/mae', 'train/mse', 'train/rse', 'train/forecast_loss', 'train/class_loss', 'train/loss', 'train/acc', 'gpu_id: 0/memory.used (MB)', 'gpu_id: 1/memory.used (MB)']. Condition can be set using `monitor` key in lr scheduler dict

**Comment by jakegrigsby** [comment]
@jeevanu @dialuser would you mind sending the training command you're using?

**Comment by jeevanu** [comment]
Not solved yet, this is command I have given, 

python train.py spacetimeformer toy2 --run_name spatiotemporal_toy2 --d_model 10 --d_ff 40 --enc_layers 4 --dec_layers 4 --batch_size 32 --start_token_len 4 --n_heads 4 --grad_clip_norm 1 --early_stopping --trials 1

**Comment by vinh-cao** [comment]
using your command I get this error:

```
 File "/media/work/Workspaces/CAO/Stocks/spacetimeformer/spacetimeformer/spacetimeformer_model/nn/attn.py", line 376, in forward
    out = out.view(B, L, -1)
RuntimeError: view size is not compatible with input tensor's size and stride (at least one dimension spans across two contiguous subspaces). Use .reshape(...) instead.
```

**Comment by jeevanu** [comment]
The command works fine on my intel pc, but getting this error on my AMD ,, Anything worng with lib

**Comment by jakegrigsby** [comment]
This was a pytorch lightning versioning issue with the third-party lr scheduler. The code was originally developed on v1.3 but I've just merged a PR that should (hopefully) fix the problem up to the current v1.5.

The second problem in this thread is caused by changing the toy dset training command from `--d_model 100 --d_ff 400` to `--d_model 10 --d_ff 40` without adjusting `--n_heads`. Breaking query, key, and value vectors with length 10 into 4 heads was creating shape issues. If you need to use a model that small it should run fine with 1 or 2 heads.
